editor/src/components/inspector/sections/layout-section/flex-element-subsection/flex-element-subsection.tsx

```diff
@@ -65,59 +72,96 @@ export const FlexElementSubsectionExperiment = betterReactMemo(
   },
 )
 
-const MainAxisControls = betterReactMemo(
-  'MainAxisControls',
-  (props: FlexElementSubsectionProps) => {
-    const isRowLayouted =
-      props.parentFlexDirection === 'row' || props.parentFlexDirection === 'row-reverse'
+export function useInitialFixedSectionState(parentFlexDirection: string | null): boolean {
+  const isRowLayouted = parentFlexDirection === 'row' || parentFlexDirection === 'row-reverse'
 
-    const width = useInspectorLayoutInfo('Width')
-    const minWidth = useInspectorLayoutInfo('minWidth')
-    const maxWidth = useInspectorLayoutInfo('maxWidth')
-    const height = useInspectorLayoutInfo('Height')
-    const minHeight = useInspectorLayoutInfo('minHeight')
-    const maxHeight = useInspectorLayoutInfo('maxHeight')
-    const alignSelf = useInspectorLayoutInfo('alignSelf')
-    const flexBasis = useInspectorLayoutInfo('flexBasis')
-    const flexGrow = useInspectorLayoutInfo('flexGrow')
-    const flexShrink = useInspectorLayoutInfo('flexShrink')
+  const width = useInspectorLayoutInfo('Width')
+  const minWidth = useInspectorLayoutInfo('minWidth')
+  const maxWidth = useInspectorLayoutInfo('maxWidth')
+  const height = useInspectorLayoutInfo('Height')
+  const minHeight = useInspectorLayoutInfo('minHeight')
+  const maxHeight = useInspectorLayoutInfo('maxHeight')
 
-    const isMainAxisVisibleForFlexDirection = isRowLayouted
-      ? isNotUnsetDefaultOrDetected(width.controlStatus) ||
+  return isRowLayouted
+    ? isNotUnsetDefaultOrDetected(width.controlStatus) ||
         isNotUnsetDefaultOrDetected(minWidth.controlStatus) ||
         isNotUnsetDefaultOrDetected(maxWidth.controlStatus)
-      : isNotUnsetDefaultOrDetected(height.controlStatus) ||
+    : isNotUnsetDefaultOrDetected(height.controlStatus) ||
         isNotUnsetDefaultOrDetected(minHeight.controlStatus) ||
         isNotUnsetDefaultOrDetected(maxHeight.controlStatus)
+}
 
-    const isMainAxisVisible =
-      isMainAxisVisibleForFlexDirection ||
-      isNotUnsetDefaultOrDetected(alignSelf.controlStatus) ||
-      isNotUnsetDefaultOrDetected(flexBasis.controlStatus) ||
-      isNotUnsetDefaultOrDetected(flexGrow.controlStatus) ||
-      isNotUnsetDefaultOrDetected(flexShrink.controlStatus)
+export function useInitialAdvancedSectionState(): boolean {
+  const alignSelf = useInspectorLayoutInfo('alignSelf')
+  return isNotUnsetDefaultOrDetected(alignSelf.controlStatus)
```

Comment:
I really struggle to mentally parse this function name. It's effectively saying "the user has explicitly given this thing a value", right?

Reply:
yes :D

Reply:
Ok I might rename this in the near future then :D

Reply:
there are a scary amount of controlstatuses